### ISSUE TITLE
Issue with upgrading sails mongo from 0.10.0-rc5 to 0.10.x

### ISSUE THREAD

**Issue by Harishgolla** [balderdashy/sails-mongo#230]
I am new to sails-mongo. We are facing below mentioned issue while trying to logging in our portal (However sails lift is successful) with 0.10.1.

error: Sent 500 ("Server Error") response
error: Error: Error updating a record.
    at /Users/admin/Desktop/NodeJs/schooner-nodemod/schooner/node_modules/sails/node_modules/waterline/lib/waterline/model/lib/defaultMethods/save.js:174:19
    at /Users/admin/Desktop/NodeJs/schooner-nodemod/schooner/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:454:17
    at /Users/admin/Desktop/NodeJs/schooner-nodemod/schooner/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:444:17
    at Array.forEach (native)
    at _each (/Users/admin/Desktop/NodeJs/schooner-nodemod/schooner/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:46:24)
    at Object.taskComplete (/Users/admin/Desktop/NodeJs/schooner-nodemod/schooner/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:443:13)
    at processImmediate [as _immediateCallback] (timers.js:345:15) [Error: Error updating a record.]

I could see explicitly this issue was from this version of sails-mongo https://github.com/balderdashy/sails-mongo/commit/cff893cdc19b9ecda711edc25cfe458f9b70acf0 (where primary criteria has been changed

User account Model used
module.exports = {
  connection: ['mongodb'],
  attributes: {
    id: {
      type: 'object',
      primaryKey: true

    },
    firstName: 'string',
    lastName: 'string',
    accountId: 'string',
    contactId: 'string',
    roles: 'array',
    showLoginMsg: 'boolean',
    communityHandle: 'string',
    password: {
      type: 'string',
      minLength: 6,
      required: true,
      columnName: 'passwordHash'
    },
    email: {
      type: 'email',
      required: true,
      unique: true
    },
    username: {
      type: 'string',
      unique: true,

    },
    groups: 'array',
    authToken: 'string',      
    registrationAuthKey: 'string',
 ...................................
}

On removing id from model, logging in is working as expected.(db.collection.find is working fine) 
Can't we perform collection.find with id as primary key?

**Comment by sailsbot** [comment]
Thanks for posting, @Harishgolla.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
